cmake/sysbuild/sign_sysbuild.cmake

```diff
@@ -0,0 +1,326 @@
+  if(NOT CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS STREQUAL "")
+    separate_arguments(imgtool_args UNIX_COMMAND ${CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS})
+  else()
+    set(imgtool_args)
+  endif()
+
+  # Fetch version and flags from the main image Kconfig.
+  set(CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION)
+  set(CONFIG_MCUBOOT_IMGTOOL_OVERWRITE_ONLY)
+  set(CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE)
+  set(CONFIG_MCUBOOT_APPLICATION_FIRMWARE_UPDATER)
+  sysbuild_get(CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION IMAGE ${main_image} VAR CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION KCONFIG)
+  sysbuild_get(CONFIG_MCUBOOT_IMGTOOL_OVERWRITE_ONLY IMAGE ${main_image} VAR CONFIG_MCUBOOT_IMGTOOL_OVERWRITE_ONLY KCONFIG)
+  sysbuild_get(CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE IMAGE ${main_image} VAR CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE KCONFIG)
+  sysbuild_get(CONFIG_MCUBOOT_APPLICATION_FIRMWARE_UPDATER IMAGE ${main_image} VAR CONFIG_MCUBOOT_APPLICATION_FIRMWARE_UPDATER KCONFIG)
+
+  string(REGEX MATCH ".*_secondary_app$" is_secondary_app ${main_image})
+
+  # Fetch devicetree details for flash and slot information.
+  dt_chosen(flash_node TARGET mcuboot PROPERTY "zephyr,flash")
+  dt_prop(write_block_size TARGET mcuboot PATH "${flash_node}" PROPERTY "write-block-size")
+  dt_nodelabel(slot0_path TARGET mcuboot NODELABEL "slot0_partition" REQUIRED)
+  dt_reg_addr(slot0_addr TARGET mcuboot PATH ${slot0_path})
+  dt_reg_size(slot0_size TARGET mcuboot PATH ${slot0_path})
+  dt_nodelabel(slot1_path TARGET mcuboot NODELABEL "slot1_partition" REQUIRED)
+  dt_reg_addr(slot1_addr TARGET mcuboot PATH ${slot1_path})
+  dt_reg_size(slot1_size TARGET mcuboot PATH ${slot1_path})
+  if(NOT write_block_size)
+    set(write_block_size 4)
+    message(WARNING "slot0_partition write block size devicetree parameter is missing, assuming write block size is 4")
+  endif()
+
+  # Fetch devicetree details for executable RAM region.
+  dt_chosen(ram_load_dev TARGET mcuboot PROPERTY "mcuboot,ram-load-dev")
+  if(DEFINED ram_load_dev)
+    dt_reg_addr(ram_load_address TARGET mcuboot PATH ${ram_load_dev})
+  else()
+    dt_chosen(chosen_ram TARGET mcuboot PROPERTY "zephyr,sram")
+    dt_reg_addr(ram_load_address TARGET mcuboot PATH ${chosen_ram})
+  endif()
+
+  # Fetch devicetree details for the active code partition.
+  dt_chosen(code_flash TARGET ${main_image} PROPERTY "zephyr,code-partition")
+  dt_reg_addr(code_addr TARGET ${main_image} PATH ${code_flash})
+  set(start_offset)
+  sysbuild_get(start_offset IMAGE ${main_image} VAR CONFIG_ROM_START_OFFSET KCONFIG)
+
+  # Append key file path.
+  if(NOT "${keyfile}" STREQUAL "")
+    set(imgtool_args --key "${keyfile}" ${imgtool_args})
+  endif()
+
+  # Construct imgtool command, based on the selected MCUboot mode.
+  set(imgtool_rom_command)
+  if(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT OR SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP)
+    if(is_secondary_app)
+      MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${slot1_addr}")
```

Comment:
not sure what any of these are doing, slot_offset is just start_offset

Reply:
I don't get it, the point was all images are merged, the the one image is signed (so one signing for combined slot 0, one signing for a combined slot 1), and this is the header offset of those images which should be a fixed size

Reply:
the app image should be first not second (and this is how images are booted, mcuboot doesn't boot image data **plus** an anount, it boots directly at the image data offset), and signing should be disabled on each image to prevent doing the same thing multiple times (or having files that users think are correct which they try to use)

Reply:
I do not understand your point of view. Yes, signing is done once and is disabled for other images.
I simply provided a more robust equation for the offset, so the user will not encounter undefined behavior if the partitions are placed in a different order - what's wrong with this approach?

Reply:
NVM, this case is either way blocked by imgtool implementation, so I'll simplify it.